dask/array/utils.py

```diff
@@ -121,6 +125,21 @@ def assert_eq(a, b, check_shape=True, check_graph=True, **kwargs):
 
     try:
         assert a.shape == b.shape
+        if check_meta:
+            if hasattr(a, '_meta') and hasattr(b, '_meta'):
+                assert a._meta.ndim == b._meta.ndim
+            if hasattr(a_original, '_meta'):
+                assert a_original._meta.ndim == a.ndim
+                if a_meta is not None:
+                    assert type(a_original._meta) == type(a_meta)
+                    if not (np.isscalar(a_meta) or np.isscalar(a_computed)):
+                        assert type(a_meta) == type(a_computed)
```

Comment:
Here I would also hope that `assert_eq(a_original._meta, normalize_meta(a_computed))` would pass and cover most of these cases.

Reply:
Perhaps that would be possible. But for now it's somewhat complex as I want to ensure we check for some not so obvious cases. For example, I want to make sure we validate meta for Sparse arrays before they're densified, after which we can't check that meta was handled properly in all stages.